### DISCUSSION TITLE
"Session not authenticated" error for public app

### DISCUSSION THREAD

**Discussion by Rootse** [Budibase/budibase#6888] [Help] [answered]
Hello. I have a problem, when I click on a button in a public application, I get the error "Session not authenticated". At the database, change this access to public, but the error remains.
The button activates the trigger, changes the row in the table and updates the data. What could be the problem?

![Screenshot_32](https://user-images.githubusercontent.com/28269608/180803379-a1fc1154-5dbb-445a-ab11-b8e2427d915f.png)
![Screenshot_31](https://user-images.githubusercontent.com/28269608/180803406-b4d4ac83-a651-4e47-bba8-203efacf6ac4.png)
![Screenshot_33](https://user-images.githubusercontent.com/28269608/180803410-cab86abc-5867-4429-92d2-e35dee232dd3.png)

**Comment by melohagan** [comment]
Hi @Rootse 

This looks like a bug. Would you be able to raise a bug here: https://github.com/Budibase/budibase/issues/new/choose 
with the reproduction steps please.

**Comment by proton029** [comment]
I have come across the same issue. The problem was that I have published my application before I changed the DB access settings. Once I have changed the settings the change wont get reflected on to the published app you have to unpublish and publish again to see the settings reflected.

**Comment by melohagan** [comment, marked as answer]
You need to ensure the minimum role of the automation trigger is set to public:

![Screenshot 2022-08-18 at 16 51 56](https://user-images.githubusercontent.com/101575380/185439310-7754c61e-411e-4316-9d11-10d0619d533c.png)
